File: src/javascript/FindTheDifference.js
function findTheDifference(s, t) {
  let sArr = s.split("");
  let tArr = t.split("");
  let sObj = {};
  let tObj = {};
  for (const element of sArr) {
    if (sObj[element]) {
      sObj[element]++;
    } else {
      sObj[element] = 1;
    }
  }
  for (const element of tArr) {
    if (tObj[element]) {
      tObj[element]++;
    } else {
      tObj[element] = 1;
    }
  }
  for (let key in tObj) {
    if (sObj[key] !== tObj[key]) {
      return key;
    }
  }
}

const s = "abcd";
const t = "abcde";
let c = findTheDifference(s, t);
console.log(c);

File: src/FindTheDifference.java
public class FindTheDifference {

  public static void main(String[] args) {

    var s = "abcd";
    var t = "abcde";
    char c = findTheDifference(s, t);
    System.out.println(c);
  }

  public static char findTheDifference(String s, String t) {
    int[] memory = new int[26];
    for (int i = 0; i < s.length(); i++) {
      memory[s.charAt(i) - 'a']++;
    }
    for (int i = 0; i < t.length(); i++) {
      memory[t.charAt(i) - 'a']--;
    }
    for (int i = 0; i < memory.length; i++) {
      if (memory[i] < 0) {
        return (char) (i + 'a');
      }
    }
    return ' ';
  }
}
